examples/all-clusters-app/linux/WindowCoveringManager.cpp

```diff
@@ -0,0 +1,138 @@
+/**
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "WindowCoveringManager.h"
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+#include <system/SystemClock.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters::WindowCovering;
+using namespace chip::System::Clock::Literals;
+
+namespace {
+
+constexpr const System::Clock::Milliseconds32 kIncrementMovementTimeout = 700_ms32;
+constexpr const uint16_t kDefaultMovementStep                           = 2000;
+
+} // namespace
+
+void WindowCoveringManager::Init(EndpointId endpoint)
+{
+    mState = OperationalState::Stall;
+    mCurrentPosition.SetNonNull(static_cast<uint16_t>(0));
+    mTargetPosition.SetNonNull(static_cast<uint16_t>(0));
+    SetEndpoint(endpoint);
+}
+
+void WindowCoveringManager::HandleMovementTimer(System::Layer * layer, void * aAppState)
+{
+    WindowCoveringManager * manager = reinterpret_cast<WindowCoveringManager *>(aAppState);
+
+    VerifyOrReturn(manager->mState != OperationalState::Stall);
+
+    uint16_t currentPosition = manager->mCurrentPosition.Value();
+    uint16_t targetPosition  = manager->mTargetPosition.Value();
+
+    ChipLogProgress(NotSpecified, "HandleMovementTimer:currentPosition:%u, targetPosition:%u", currentPosition, targetPosition);
+
+    if (OperationalState::MovingUpOrOpen == manager->mState)
+    {
+        if (currentPosition > targetPosition)
```

Comment:
there is a function to compare already available i would have prefer keeping using it

Reply:
ack